### ISSUE TITLE
[Schema Inaccuracy] Commit SHA and URL are nullable

### ISSUE THREAD

**Issue by Maxim-Mazurok** [github/rest-api-description#114]
# Schema Inaccuracy

In [api.github.com.json](https://github.com/github/rest-api-description/blob/015ac73e48d69e9b2ee968682be32ab03865e5a2/descriptions/api.github.com/api.github.com.json), lines 57297- 57312, I can see this:
```json5
{
  /*...*/
  "commit": {
    "title": "Commit",
    "description": "Commit",
    "type": "object",
    "properties": {
      "url": {
        "type": "string",
        "format": "uri",
        "example": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
        "nullable": true /* <<<=== */
      },
      "sha": {
        "type": "string",
        "example": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
        "nullable": true /* <<<=== */
      }
      /*...*/
    }
    /*...*/
  }
  /*...*/
}
```

Which [later](https://github.com/octokit/openapi-types.ts/blob/d891b1a941ab4f645a5e37d416033fe1ea9ceae6/generated/types.ts#L25204-L25206) results in:
```ts
commit: {
  url: string | null;
  sha: string | null;
  //...
}
```

So, my code:
```ts
const branchData = await octokit.repos.getBranch({
  owner,
  repo,
  branch,
});

const sha: string = branchData.data.commit.sha;
```
Fails to compile with the error:
```
Type 'string | null' is not assignable to type 'string'.
  Type 'null' is not assignable to type 'string'.ts(2322)
```

## Expected

`url` and `sha` properties on `commit` schema should be of type `string` instead of `string | null`

## Reproduction Steps

N/A (or see my code example above)

**Comment by gr2m** [comment]
I've ran into the same problem yesterday:
https://github.com/googleapis/repo-automation-bots/pull/1214/files#diff-b9ae098571fb2b76f6b8b865ab001120a6f9b65aa3f1e7f916b4d739bdb1b5edR104

I worked around it because I wasn't sure if there is valid case when `commit.sha` and be `null`. Is there?

The `sha` and `url` keys for the commit object are defined in the OpenAPI spec in `components/schema/commit`

```js
      "commit": {
        "title": "Commit",
        "description": "Commit",
        "type": "object",
        "properties": {
          "url": {
            "type": "string",
            "format": "uri",
            "example": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
            "nullable": true
          },
          "sha": {
            "type": "string",
            "example": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
            "nullable": true
          },
          "node_id": {
            "type": "string",
            "example": "MDY6Q29tbWl0NmRjYjA5YjViNTc4NzVmMzM0ZjYxYWViZWQ2OTVlMmU0MTkzZGI1ZQ=="
          },
          "html_url": {
            "type": "string",
            "format": "uri",
            "example": "https://github.com/octocat/Hello-World/commit/6dcb09b5b57875f334f61aebed695e2e4193db5e"
          },
          "comments_url": {
            "type": "string",
            "format": "uri",
            "example": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e/comments"
          },
          "commit": {
            "type": "object",
            "properties": {
              "url": {
                "type": "string",
                "format": "uri",
                "example": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e"
              },
              "author": {
                "nullable": true,
                "allOf": [{ "$ref": "#/components/schemas/git-user" }]
              },
              "committer": {
                "nullable": true,
                "allOf": [{ "$ref": "#/components/schemas/git-user" }]
              },
              "message": { "type": "string", "example": "Fix all the bugs" },
              "comment_count": { "type": "integer", "example": 0 },
              "tree": {
                "type": "object",
                "properties": {
                  "sha": {
                    "type": "string",
                    "example": "827efc6d56897b048c772eb4087f854f46256132"
                  },
                  "url": {
                    "type": "string",
                    "format": "uri",
                    "example": "https://api.github.com/repos/octocat/Hello-World/tree/827efc6d56897b048c772eb4087f854f46256132"
                  }
                },
                "required": ["sha", "url"]
              },
              "verification": { "$ref": "#/components/schemas/verification" }
            },
            "required": [
              "author",
              "committer",
              "comment_count",
              "message",
              "tree",
              "url"
            ]
          },
          "author": {
            "nullable": true,
            "allOf": [{ "$ref": "#/components/schemas/simple-user" }]
          },
          "committer": {
            "nullable": true,
            "allOf": [{ "$ref": "#/components/schemas/simple-user" }]
          },
          "parents": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "sha": {
                  "type": "string",
                  "example": "7638417db6d59f3c431d3e1f261cc637155684cd"
                },
                "url": {
                  "type": "string",
                  "format": "uri",
                  "example": "https://api.github.com/repos/octocat/Hello-World/commits/7638417db6d59f3c431d3e1f261cc637155684cd"
                },
                "html_url": {
                  "type": "string",
                  "format": "uri",
                  "example": "https://github.com/octocat/Hello-World/commit/7638417db6d59f3c431d3e1f261cc637155684cd"
                }
              },
              "required": ["sha", "url"]
            }
          },
          "stats": {
            "type": "object",
            "properties": {
              "additions": { "type": "integer" },
              "deletions": { "type": "integer" },
              "total": { "type": "integer" }
            }
          },
          "files": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "filename": { "type": "string" },
                "additions": { "type": "integer" },
                "deletions": { "type": "integer" },
                "changes": { "type": "integer" },
                "status": { "type": "string" },
                "raw_url": { "type": "string" },
                "blob_url": { "type": "string" },
                "patch": { "type": "string" },
                "sha": {
                  "type": "string",
                  "example": "\"1e8e60ce9733d5283f7836fa602b6365a66b2567\""
                },
                "contents_url": {
                  "type": "string",
                  "example": "\"https://api.github.com/repos/owner-3d68404b07d25daeb2d4a6bf/AAA_Public_Repo/contents/geometry.js?ref=c3956841a7cb7e8ba4a6fd923568d86958f01573\""
                },
                "previous_filename": {
                  "type": "string",
                  "example": "\"subdir/before_name.txt\""
                }
              }
            }
          }
        },
        "required": [
          "url",
          "sha",
          "node_id",
          "html_url",
          "comments_url",
          "commit",
          "author",
          "committer",
          "parents"
        ]
      },
```

**Comment by Maxim-Mazurok** [comment]
> I wasn't sure if there is valid case when commit.sha and be null. Is there?

Same question. I guess that sometimes it can be null, but for my use-case - I can't see how it can be null. When I'm getting branch - it should have commit. If there's no commit - there can be no branch. Right?

_Off-topic: looks like your project is also related to Google APIs, yay 😀_

**Comment by gr2m** [comment]
> Off-topic: looks like your project is also related to Google APIs, yay 😀

Not directly, I'm maintaining the @octokit JS libraries. We had a big TypeScript update through which we started to consume the OpenAPI spec from `github/rest-api-description`, and the update at https://github.com/googleapis/repo-automation-bots/pull/1195 surfaced several regressions and legit problems with GitHub's OpenAPI spec along the way, so I helped the Google team to address the problems in this case

**Comment by Maxim-Mazurok** [comment]
Would be great to have @imwiss input on this one :)

**Comment by imwiss** [comment]
> Would be great to have @imwiss input on this one :)

Thanks for flagging this. I agree that I cannot think of a valid case when `commit.sha` and `commit.url` can be null, but I'd have to dig deeper to confirm.

We'll be diving into this issue in the very near future, as well as potentially other nullable fields that are causing problems. We'll make sure to post an update here as soon as we do.

In the meantime, did the workaround proposed by @gr2m unblock you?

**Comment by Maxim-Mazurok** [comment]
Awesome, thank you :)

I decided not to apply any workarounds because the old version of the API package seems to work just fine and I don't want to add a workaround only to delete it later. So, I'm not really blocked because everything works, I'm just going to stick to @octokit/rest v18.0.9 for now.

**Comment by Maxim-Mazurok** [comment]
Hello @lucascosti, any reason to close this one?... There are no changes to the schema in the main branch regarding this issue.

**Comment by lucascosti** [comment]
👋 @Maxim-Mazurok This issue was just fixed in an internal PR by @martin389 that I just merged, which automatically closed this issue. The changes will be automatically propagated out into this public repo in the next day or so. Stay tuned! 🙂

**Comment by Maxim-Mazurok** [comment]
Oh, that's great to know, thank you and Martin!

**Comment by Maxim-Mazurok** [comment]
@gr2m FYI, it's now fixed in 18.0.14 🎉
